desc/builder/file.go

```diff
@@ -573,11 +573,15 @@ func (fb *FileBuilder) TryAddService(sb *ServiceBuilder) error {
 //
 // Knowledge of custom options can also be provided by using BuildOptions with
 // an ExtensionRegistry, when building the file.
+// This function panics If builder for same proto file are added twice.
 func (fb *FileBuilder) AddDependency(dep *FileBuilder) *FileBuilder {
 	if fb.explicitDeps == nil {
-		fb.explicitDeps = map[*FileBuilder]struct{}{}
+		fb.explicitDeps = map[string]*FileBuilder{}
 	}
-	fb.explicitDeps[dep] = struct{}{}
+	if _, exists := fb.explicitDeps[dep.GetName()]; exists {
```

Comment:
The old code did not fail if you tried to explicitly add the same dependency twice. But now the second time will panic. So I think we can tweak this to reproduce the old behavior:

```go
if prev, exists := fb.explicitDeps[dep.GetName()]; exists && prev != dep  {
```

Also, I just realized another snag: a `*FileBuilder` does not have to have a name. You can create an "anonymous" file and the name will be generated when it is actually built. Also, the name is mutable, which means it could be changed _after_ adding it to this map, which would be problematic.

So I apologize for suggesting this check. But I think we instead have to leave this the way it was, and just track builders by their pointer address. Then, during `Build`, there could be an explicit check and thus a better error message provided when there are duplicate file names. And since it's also possible there could be something with the same name over in the descriptor map, we need that check anyway to catch duplicates where one dependency was provided as a builder and another with the same name as a descriptor.

Reply:
Yeah, looks we have to validate during `Build` only. Made the changes accordingly.

Sorry for delay in updates.